decoder/stats.go

```diff
@@ -198,15 +205,26 @@ func updateEncounterStats(pokemon *Pokemon) {
 		pokemonStatsLock.Lock()
 		defer pokemonStatsLock.Unlock()
 
-		countStats := pokemonCount[areaName]
-		if countStats == nil {
-			countStats = &areaPokemonCountDetail{}
+		countStats, exists := pokemonCount[areaName]
+		if !exists {
+			countStats = &areaPokemonCountDetail{
+				hundos:      make(map[pokemonForm]int),
+				nundos:      make(map[pokemonForm]int),
+				count:       make(map[pokemonForm]int),
+				ivCount:     make(map[pokemonForm]int),
+				shinyChecks: make(map[pokemonForm]shinyChecks),
+			}
 			pokemonCount[areaName] = countStats
 		}
-		countStats.shinyChecks[pokemon.PokemonId].total++
+
+		pf := pokemonForm{pokemonId: pokemon.PokemonId, formId: formId}
+
+		entry := countStats.shinyChecks[pf]
+		entry.total++
```

Comment:
The pattern that was present before is slightly more optimal (ie changing total, and second lookup if shiny) vs the form you landed on with two lookups always. I'm not sure why you changed it?

Reply:
Readability.

Reply:
It's less readable now

Reply:
Opinion vs opinion. I will change it to get over it :)

Reply:
You are copying value types needlessly, when this change is made I'll be happy that it at least is no worse than it was (except maps being inherently slower than arrays but this is unavoidable)

Reply:
Can't change it anyway, leaving for future ref:
```go
countStats.shinyChecks[pf].total++
```
> cannot assign to struct field countStats.shinyChecks[pf].total in mapcompiler[UnaddressableFieldAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#UnaddressableFieldAssign)